### ISSUE TITLE
[Feature Request]: Collect extra-manifests recursive

### ISSUE THREAD

**Issue by markusheinemann** [kube-hetzner/terraform-hcloud-kube-hetzner#1313]
### Background

I want to deploy multiple initial workloads on my new cluster after creation. They need to be deployed in different namespaces. This can be archived with a kustomize feature by using a [common base](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/multibases/multi-namespace.md).  So I leveraged a folder structure for my `extra-manifests`:

```
.
├── extra-manifests/
│   ├── app1/
│   │   └── kustomization.yaml.tpl
│   ├── app2/
│   │   └── kustomization.yaml.tpl
│   └── kustomization.yaml.tpl
└── kube.tf
```

The purpose of `extra-manifests/kustomization.yaml.tpl` is to import `extra-manifests/app1/kustomization.yaml.tpl` and `extra-manifests/app2/kustomization.yaml.tpl`

### Issue

Unfortunately, kube-hetzner does not copy subfolders from `extra-manifests` to the nodes which results in an error when executing the custom manifests:

```
error: accumulating resources: accumulation err='accumulating resources from 'app1': evalsymlink failure on '/var/user_kustomize/app1' : lstat /var/user_kustomize/app1: no such file or directory': must build at directory: not a valid directory: evalsymlink failure on '/var/user_kustomize/app1' : lstat /var/user_kustomize/app1: no such file or directory
```

### Proposal

Recursive consideration of user-defined kubernetes manifests so that files from subfolders are copied correctly. 

I think this feature can be implemented very easily by updating the respective fileset to`fileset("extra-manifests", "**/*.yaml.tpl")`:

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/d1ba7083c6698c0cdbd7d26a548f267afd0757ad/kustomization_user.tf#L2

**Comment by markusheinemann** [comment]
I will test the changes and create PR to validate the solution

**Comment by markusheinemann** [comment]
Solved by #1314
